registry/crypto/functions.go

```diff
@@ -0,0 +1,449 @@
+package crypto
+
+import (
+	"bytes"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/dsa"
+	"crypto/ecdsa"
+	"crypto/ed25519"
+	"crypto/elliptic"
+	"crypto/hmac"
+	cryptorand "crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/binary"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+
+	bcrypt_lib "golang.org/x/crypto/bcrypt"
+	"golang.org/x/crypto/scrypt"
+)
+
+// Bcrypt generates a bcrypt hash from the given input string.
+//
+// input - the string to be hashed.
+// Returns the bcrypt hash as a string.
+//
+// Example:
+//
+//	{{ "Hello World" | bcrypt }} // Output: "$2a$12$C1qL8XVjIuGKzQXwC6g6tO"
+func (ch *CryptoRegistry) Bcrypt(input string) string {
+	hash, err := bcrypt_lib.GenerateFromPassword([]byte(input), bcrypt_lib.DefaultCost)
+	if err != nil {
+		return fmt.Sprintf("failed to encrypt string with bcrypt: %s", err)
```

Comment:
Here and everywhere

Why not wrapping the error?


```suggestion
		return fmt.Sprintf("failed to encrypt string with bcrypt: %w", err)
```

Reply:
The first reason is that "the crypto registry hasn't changed; it's the same as in Sprig currently."

The second reason is that fmt.Sprintf does not support the error-wrapping directive %w, and many functions don't properly return errors. This is one of the main reasons why I want to evolve Sprig.

Reply:
Oh I miss read and thought the signature was an error, so I thought it was an fmt.Errorf

Reply:
Don't worry the first time I try to touch that, I do the same mistake 👬